src/Dotnet.Script.Core/NuGet/NuGetMetadataReferenceResolver.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Dotnet.Script.Core.NuGet
+{
+    /// <summary>
+    /// A <see cref="MetadataReferenceResolver"/> decorator that handles
+    /// references to NuGet packages in scripts.  
+    /// </summary>
+    public class NuGetMetadataReferenceResolver : MetadataReferenceResolver
```

Comment:
Once you get the next version out the door, maybe we can look at merging this PR. I am really excited to see how this behaves as a .NetCore2.0 app.